lib/json_api_client/resource.rb

```diff
@@ -449,7 +462,7 @@ def destroy
         fill_errors
         false
       else
-        self.attributes.clear
+        mark_as_destroyed!
         self.relationships.attributes.clear
```

Comment:
if we don't want to clear attributes after `destroy`, why we should clear relationships?